### PR TITLE
change AmqpVersion to struct

### PR DESCRIPTION
**Pull request by bollhals**
## Proposed Changes

Simple small change to change it to a struct. Avoids allocation.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [x] Cosmetic change (whitespace, formatting, etc)

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

## Further Comments

I've let resharper implement the equality members, which also changed the GetHashCode. But as the type is internal, this is not breaking.

### CODE REVIEW

File: projects/RabbitMQ.Client/client/impl/AmqpVersion.cs [hunk 1]
```diff
@@ -70,27 +72,36 @@ public AmqpVersion(int major, int minor)
...
         /// <summary>
         /// Implement hashing as for value-equality.
         /// </summary>
         public override int GetHashCode()
         {
-            return 31*Major.GetHashCode() + Minor.GetHashCode();
+            unchecked
+            {
+                return (Major * 397) ^ Minor;
+            }
```
Comment (sungam3r) [inline projects/RabbitMQ.Client/client/impl/AmqpVersion.cs:104]:
```suggestion
            return (Major, Minor).GetHashCode();
```

**Review by michaelklishin — APPROVED** [review]

File: projects/RabbitMQ.Client/client/impl/AmqpVersion.cs (hunk 1 above)
Comment (michaelklishin) [reply projects/RabbitMQ.Client/client/impl/AmqpVersion.cs:104]: That seems to require an extra dependency on `System.Tuple_2`, though.

Comment (michaelklishin) [reply projects/RabbitMQ.Client/client/impl/AmqpVersion.cs:104]: Thank you for the suggestion, it is definitely a nicer option but I'm not sure if it's worth an extra dependency (if it does require one, I'm unable to compile this on .NET Core 3.1 on MacOS as is).

Comment (sungam3r) [reply projects/RabbitMQ.Client/client/impl/AmqpVersion.cs:104]: ok

Comment (bollhals) [reply projects/RabbitMQ.Client/client/impl/AmqpVersion.cs:104]: Due to .NET 4.6.1 this syntax isn't yet possible to use without additional dependencies. (Similar to that we can't use anonymous (int, int) return values)

**Pull request merged**